### PR TITLE
Variable as script

### LINKED ISSUE #75: creation of script in heroku and sending it as a variable
**I'm submitting a ...**  (check one with "x")
- [ ] bug report
- [x] feature request
- [ ] support request => Please do not submit support requests here, instead ask your query in out Gitter channel at https://gitter.im/fossasia/meilix

we are currently sending different variables for each change at the end we may have too many which will be difficult to set as env. variables for new Heroku dynos or Travis tests so it's better to create a script inside Heroku and send it as a variable to Travis

### PR DESCRIPTION
**Pull request by xeon-zolt**
#### Checklist

- [x] I have read the [Contribution & Best practices Guide](https://blog.fossasia.org/open-source-developer-guide-and-best-practices-at-fossasia) and my PR follows them.
- [x] There is a corresponding issue for this pull request.
- [x] My branch is up-to-date with the Upstream `master` branch.
- [x] The unit tests pass locally with my changes (provide a screenshot or link for test) 
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have added necessary documentation (if appropriate)

#### Short description of what this resolves:
by using this change we will be able to send a generated script as a variable to the travis to use under chroot 

#### Changes proposed in this pull request:

- added a script that will be generated/edited by webapp
- changes in app.py to set script as env variavle
- added a new variable for travis and heroku for the script
- removed unused imports
- used base64 to make a variable for travis ci

Fixes #75

### CODE REVIEW

File: app.py [hunk 1]
```diff
@@ -44,6 +44,10 @@ def index():
 				TRAVIS_TAG = urlify(TRAVIS_TAG)#this will fix url issue
 				os.environ["TRAVIS_TAG"] = TRAVIS_TAG
 				os.environ["event_url"] = event_url
+				file = open('travis_script_1.sh', 'r')
```
Comment (tabesin) [inline app.py]: I recommend to use a less generic name for the file handle. But that is just style.

File: script.sh [hunk 2]
```diff
@@ -13,12 +13,11 @@ do
...
                     \"TRAVIS_TAG\":\"${TRAVIS_TAG}\",
-                    \"event_url\":\"${event_url}\"
+                    \"event_url\":\"${event_url}\",
+                    \"TRAVIS_SCRIPT\":\"${TRAVIS_SCRIPT}\"
                 }
             }
     }}"
-    echo "This link will be ready in approx 20 minutes"
-    echo "https://github.com/xeon-zolt/meilix/releases/download/${TRAVIS_TAG}/meilix-zesty-`date +%Y%m%d`-i386.iso"
     echo "Sending Data To travis To build ISO"
```
Comment (tabesin) [inline script.sh]: Sending Data To Travis To Build An ISO Image.

?? Mind the capitalizations, many options but be consistent.

File: travis_script_1.sh [hunk 3]
```diff
@@ -0,0 +1,2 @@
+#!/usr/bin/env bash
+echo heloo
```
Comment (tabesin) [inline travis_script_1.sh]: Just a stub I assume

File: script.sh (hunk 2 above)
Comment (xeon-zolt) [inline script.sh]: let me remove them as they are not required anymore

File: app.py [hunk 4]
```diff
@@ -1,10 +1,10 @@
 import os
-from flask import Flask, render_template, Response ,request ,redirect ,url_for ,send_from_directory
```
Comment (xeon-zolt) [inline app.py]: response is not used

File: app.py [hunk 5]
```diff
@@ -1,10 +1,10 @@
...
-
+from flask import Flask, render_template ,request ,redirect ,url_for ,send_from_directory
 from werkzeug import secure_filename
-import time
-import subprocess
```
Comment (xeon-zolt) [inline app.py]: time and subprocess are also not used

File: app.py [hunk 6]
```diff
@@ -44,17 +44,20 @@ def index():
 				TRAVIS_TAG = urlify(TRAVIS_TAG)#this will fix url issue
 				os.environ["TRAVIS_TAG"] = TRAVIS_TAG
 				os.environ["event_url"] = event_url
+				with open('travis_script_1.sh','rb') as f:
+					os.environ["TRAVIS_SCRIPT"] = str(base64.b64encode(f.read()))[1:]
```
Comment (xeon-zolt) [inline app.py]: we can use it for sending wallpaper also as current system we host wallpaper due to which more than one user cannot use the webapp simultaneously

File: travis_script_1.sh (hunk 3 above)
Comment (xeon-zolt) [inline travis_script_1.sh]: to be used later when we add more features to webapp without adding more variables but sending just a script to be executed inside chroot for making changes

Comment (tabesin) [inline travis_script_1.sh]: Thank you. Please add later a line that explains the purpose of the dummy script.

**Comment by tabesin** [comment]
Are you sure this goes into the Master branch?

**Pull request merged**